### PR TITLE
Fix error strings, slice declaration and expression 'c <= 'ÿ'' is always 'true'

### PR DESCRIPTION
**Pull request by aixeshunter**
1. Error string should not be capitalized or end with punctuation, fix in `storage/remote/codec.go`.
2. Slice declaration with empty literal initializer instead of nil, change the declaration to avoid an unnecessary allocation
3.Expression 'c <= 'ÿ'' is always 'true', delete them in `pkg/textparse/promlex.l.go`.

### CODE REVIEW

File: storage/remote/read_test.go [hunk 1]
```diff
@@ -122,12 +123,12 @@ func TestSeriesSetFilter(t *testing.T) {
 			toRemove: model.LabelSet{"foo": "bar"},
 			in: &prompb.QueryResult{
 				Timeseries: []*prompb.TimeSeries{
-					{Labels: labelsToLabelsProto(labels.FromStrings("foo", "bar", "a", "b")), Samples: []prompb.Sample{}},
+					{Labels: labelsToLabelsProto(labels.FromStrings("foo", "bar", "a", "b")), Samples: samples},
```
Comment (codesome) [inline storage/remote/read_test.go]: Instead of this, `Samples: nil` would work.

Comment (aixeshunter) [reply storage/remote/read_test.go]: haha, yes, it would work.

File: pkg/textparse/promlex.l.go [hunk 2]
```diff
@@ -386,7 +386,7 @@ yystate34:
 	switch {
 	default:
 		goto yyrule17
-	case c >= '\x01' && c <= '\b' || c >= '\v' && c <= '\x1f' || c >= '!' && c <= 'z' || c >= '|' && c <= 'ÿ':
+	case c >= '\x01' && c <= '\b' || c >= '\v' && c <= '\x1f' || c >= '!' && c <= 'z' || c >= '|':
```
Comment (fabxc) [inline pkg/textparse/promlex.l.go]: This is generated code and must not be changed.

**Comment by fabxc** [comment]
As @codesome said, this will be overwritten with the next run of the generator. You could try to fix this in the generating library. But I doubt it has any notable impact on performance.

File: storage/remote/read_test.go (hunk 1 above)
Comment (juliusv) [reply storage/remote/read_test.go]: Or even just omitting `Samples` completely, in which case it will take its zero value `nil` automatically.

**Comment by codesome** [comment]
The fix for error string is already upstream and the other changes are not relevant. Hence closing this.

**Pull request closed without merging**